eth/downloader/downloader.go

```diff
@@ -1352,6 +1322,11 @@ func (d *Downloader) fetchNodeData() error {
 		deliver = func(packet dataPack) (int, error) {
 			start := time.Now()
 			return d.queue.DeliverNodeData(packet.PeerId(), packet.(*statePack).states, func(err error, delivered int) {
+				// If the peer gave us nothing, stalling fast sync, drop
+				if delivered == 0 {
```

Comment:
Actually, this check was added because Parity nodes were/are advertising eth/63 but not actually implementing it. So for the node requests they plain returned empty replies, which stalled syncing quite a lot as we always had to wait for parity RTTs to finish before pulling from someone who actually had the data. Doing this for every batch of state entries significantly stalled sync time.

We might reconsider this a bit later down the line I agree, but I'd rather not go into this direction with this PR as it's not *that* trivial as to how to handle potential attack vectors.